File: C++/Tree Alogrithms/level_order_traversal.cpp
#include<iostream>
#include<bits/stdc++.h>

using namespace std;

typedef struct node{
    node* right;
    node* left;
    int val;
}node;

//inorder traversal root -> left ->  right 
//recursive

node* createNode(int data){
    node* newNode = new node();
    newNode -> right = NULL;
    newNode -> left = NULL;
    newNode -> val = data;
    return newNode;
}


void levelorderTraversal(node* root){
    if(root == NULL)
        return;

    queue<node* > q;
    q.push(root);

    while(q.size() > 0){
        node* front = q.front();
        q.pop();
        cout<<front->val<<" ";

        if(front -> left)
            q.push(front -> left);
        if(front -> right)
            q.push(front -> right);
    }
}


int main(){

    node* root = NULL;
    root = createNode(5);
    root -> left = createNode(3);
    root -> left -> left = createNode(1);
    root -> left -> right = createNode(4);

    root -> right = createNode(7);
    root -> right -> right = createNode(9);
    root -> right -> left = createNode(6);

    levelorderTraversal(root);

    return 0;
}

File: Python/Tree Algorithms/level_order_traversal.py
# Python program for implementation of Level Order Traversal

# Structure of a node
class Node:
    def __init__(self ,key):
        self.data = key
        self.left = None
        self.right = None
        
# print level order traversal
def printLevelOrder(root):
    if root is None:
        return
# create an empty node_queue for Level order Traversal   
    node_queue = [] 
    node_queue.append(root)
    
#print Front of the queue and pop it from the queue
    while(len(node_queue) > 0):
        print (node_queue[0].data)
        node = node_queue.pop(0)
        
# Enqueue left child
        if node.left is not None:
            node_queue.append(node.left)
            
# Enqueue right child
        if node.right is not None:
            node_queue.append(node.right)
 
root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

print ("Binary tree's level order traversal is :")
printLevelOrder(root)
